### ISSUE TITLE
Order item cache potential discrepencies

### ISSUE THREAD

**Issue by helgatheviking** [woocommerce/woocommerce#31656]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

The [order data store](https://github.com/woocommerce/woocommerce/blob/3611d4643791bad87a0d3e6e73e031bb80447417/plugins/woocommerce/includes/data-stores/abstract-wc-order-data-store-cpt.php#L379) fetches the `order_item_type`, `order_item_id` , `order_id`, and `order_item_name`  when querying the order items for a particular order.

Each found row is then cached as the item. but then when the individual item is read, only the `order_id`, and `order_item_name` are [queried and cached](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/data-stores/abstract-wc-order-item-type-data-store.php#L130).

Is there a reason to cache different values depending on how the data is fetched?

I wasn't sure if this was a bug, but it seemed strange to me that the cached values would be different.

### Expected behavior

I would expect to cache the same values.

### Actual behavior

Some fields are cached when the order Items are read in the order data store. If an individual order item is read, different values are cached.

### Steps to reproduce

n/a

### WordPress Environment

n/a

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

**Comment by lanej0** [comment]
Hi @helgatheviking,

Thank you for opening the issue! It requires further feedback from the WooCommerce Core team. I am adding the `needs developer feedback` label to this issue so that the Core team could take a look. 

Please note it may take a few days for them to get to this issue. Thank you for your patience.

**Comment by vedanshujain** [comment]
@helgatheviking good catch. this does looks like an inconsistency that should be fixed.

**Comment by helgatheviking** [comment]
@vedanshujain which things _should_ get cached? My guess is the one in the order data store is correct and the order item data store needs updating? Or is it the inverse?

**Comment by pmcpinto** [comment]
Thanks for the report.

I know this has been open for a very long time, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

I've taken a look and marked this ready for the engineering team to pick up.
